docs/hoon/library/2k.md

```diff
@@ -180,164 +174,192 @@ Source
         ?:  |(?=($~ q.b) (vor n.a n.q.b))
           [n.a l.a q.b]
         [n.q.b [n.a l.a l.q.b] r.q.b]
-      ::
 
-Examples
---------
 
-    > =s (~(gas to *(qeu @)) `(list @)`~[1 2 3])
+#### Examples
+
+    > =s (~(gas to *(tree @)) `(list @)`~[1 2 3])
     > ~(get to s)
-    [p=1 q={3 2}]
+    [n=2 l={3} r={1}]
+
     > ~(get to ~)
     ! exit
 
-------------------------------------------------------------------------
-
+***
 ### `+-nap:to`
 
-Remove last element 
+Remove head
 
-Removes the head of `++queue` `a`, producing the resulting queue.
+Removes the head of queue `a`, producing the resulting queue.
 
-Accepts
--------
+#### Accepts
 
 `a` is a queue.
 
-Produces
---------
+#### Produces
 
 A queue.
 
-Source
-------
+#### Source
 
-      +-  nap                                               ::  removes head
+      +-  nap                                              
         ?>  ?=(^ a)
-        ?:  =(~ l.a)  r.a
+        ?~  l.a  r.a
         =+  b=get(a l.a)
         bal(a ^+(a [p.b q.b r.a]))
-      ::
 
-Examples
---------
+#### Examples
 
     > =a (~(gas to `(qeu @)`~) `(list @)`[1 2 3 4 5 6 ~])
     > -.a
     n=6
+
     > =b ~(nap to a)
     > -.b
     n=2
+
+    > a
+    [n=6 l={} r={5 4 3 2 1}]
+
     > b
+    [n=2 l=[n=4 l={5} r={3}] r=[n=1 l={} r={}]]
+
+    > `(qeu)`b
     {5 4 3 2 1}
-    > a
+
+    > `(qeu)`a
     {6 5 4 3 2 1}
 
-------------------------------------------------------------------------
+***
+###  `+-nip:to`
+
+Removes the root of queue `a`, producing the resulting queue.
```

Comment:
What significance does the root element have? I wonder if this is really just part of the private interface for the queue to rebalance itself, as opposed to something intended to be called publicly. If so, we should mention that in the documentation.